agent_api/src/main/java/dev/aikido/agent_api/storage/routes/RoutesStore.java

```diff
@@ -0,0 +1,65 @@
+package dev.aikido.agent_api.storage.routes;
+
+import dev.aikido.agent_api.api_discovery.APISpec;
+import dev.aikido.agent_api.context.RouteMetadata;
+import dev.aikido.agent_api.helpers.logging.LogManager;
+import dev.aikido.agent_api.helpers.logging.Logger;
+
+import java.util.concurrent.locks.ReentrantLock;
+
+public final class RoutesStore {
+    private static final Logger logger = LogManager.getLogger(RoutesStore.class);
+    private static final ReentrantLock mutex = new ReentrantLock();
+    private static final Routes routes = new Routes();
+
+    private RoutesStore() {
+
+    }
+
+    public static int getRouteHits(RouteMetadata routeMetadata) {
+        mutex.lock();
+        int hits = routes.get(routeMetadata).getHits();
+        mutex.unlock();
+        return hits;
+    }
+
+    public static RouteEntry[] getRoutesAsList() {
+        mutex.lock();
+        RouteEntry[] routesList = routes.asList();
+        mutex.unlock();
+        return routesList;
+    }
+
+    public static void updateApiSpec(RouteMetadata routeMetadata, APISpec apiSpec) {
+        mutex.lock();
+        try {
+            RouteEntry route = routes.get(routeMetadata);
```

Comment:
Also nice candidate for `Optional` later?

Reply:
Would be some more global changes I think, would want to keep those separate since this is already huge haha